crates/ark/src/interface.rs

```diff
@@ -1059,7 +1088,13 @@ impl RMain {
                 if self.dap.is_debugging() {
                     let continue_cmds = vec!["n", "f", "c", "cont"];
                     if continue_cmds.contains(&&code[..]) {
+                        // We're stepping so we want to focus the next location we stop at
+                        self.debug_preserve_focus = false;
                         self.dap.send_dap(DapBackendEvent::Continued);
+                    } else {
+                        // The user is evaluating some other expression so preserve current focus
+                        // https://github.com/posit-dev/positron/issues/3151
+                        self.debug_preserve_focus = true;
```

Comment:
```suggestion
                        // We're stepping so we want to focus the next location we stop at
                        self.dap.set_preserve_focus(false);
                        self.dap.send_dap(DapBackendEvent::Continued);
                    } else {
                        // The user is evaluating some other expression so preserve current focus
                        // https://github.com/posit-dev/positron/issues/3151
                        self.dap.set_preserve_focus(true);
```

I'm imagining you'd then expose this like this